x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/UpdateConnectorConfigurationAction.java

```diff
@@ -37,13 +35,13 @@
 import static org.elasticsearch.action.ValidateActions.addValidationError;
 import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
 
-public class UpdateConnectorConfigurationAction extends ActionType<UpdateConnectorConfigurationAction.Response> {
+public class UpdateConnectorConfigurationAction extends ActionType<ActionResponse.Empty> {
 
     public static final UpdateConnectorConfigurationAction INSTANCE = new UpdateConnectorConfigurationAction();
     public static final String NAME = "cluster:admin/xpack/connector/update_configuration";
 
     public UpdateConnectorConfigurationAction() {
-        super(NAME, UpdateConnectorConfigurationAction.Response::new);
+        super(NAME, in -> ActionResponse.Empty.INSTANCE);
```

Comment:
Are these actions invoked remotely? If not, use `org.elasticsearch.common.io.stream.Writeable.Reader#localOnly` here. Or, even better, drop the subclasses of `ActionType` in favour of `org.elasticsearch.action.ActionType#localOnly` - it's a common pattern (it made more sense in the distant past) but we're trying to remove it (see e.g. https://github.com/elastic/elasticsearch/pull/103055, https://github.com/elastic/elasticsearch/pull/102800, https://github.com/elastic/elasticsearch/pull/102354, https://github.com/elastic/elasticsearch/pull/101053, https://github.com/elastic/elasticsearch/pull/101037). That can be a follow-up tho.

Reply:
> Are these actions invoked remotely?

@DaveCTurner The actions in question are designed for remote invocation, such as by a user managing a connector document's state via Kibana or through the connectors framework. Given this context, could you share any best practice recommendations for handling the response representation of these actions? I hope my understanding of "remotely" w.r.t. your question is accurate.

Regarding the use of `ActionResponse.Empty`, I agree that it offers an elegant solution, as we can leverage the status code to indicate success or error. The initial choice of `AcknowledgedResponse` was to explicitly confirm in the response payload that the update operation was successful. However, I'm contemplating whether a user-facing API should be more explicit in confirming success, particularly when using tools like the Kibana dev console or curl.

Reply:
Sorry, I'll clarify - the user's interactions with the REST layer are certainly "remote", but the REST layer invokes these _transport_ actions on the local node only so there's no need to specify. You only need to specify a response reader if you're invoking these actions via the client in a different node (which in practice means you're invoking them on a remote cluster). I don't think that's happening here is it?

> However, I'm contemplating whether a user-facing API should be more explicit in confirming success, particularly when using tools like the Kibana dev console or curl.

The representation of the eventual JSON response is under your control, and is somewhat independent of the response type of the transport action. I'd discourage using `"acknowledged": true` since that will be confusing to users that are familiar with what `AcknowledgedResponse` means, but you could certainly return `{"result":"updated"}` or similar.

Reply:
Thank you for clarification. 👍  We don't invoke those actions via a client in a different node. Handling those actions results in updates to documents `connector index` (CRUD operations on the index via the client). This API is essentially just a wrapper around CRUD index operations with certain validations. 

> The representation of the eventual JSON response is under your control, and is somewhat independent of the response type of the transport action. I'd discourage using "acknowledged": true since that will be confusing to users that are familiar with what AcknowledgedResponse means, but you could certainly return {"result":"updated"} or similar.

Makes sense, I think we have a good understanding now :) So it seems a viable approach would be to create a single class e.g. `ConnectorUpdateActionResponse` that would return for example`{"result":"updated"}` to the user and would be used by all update API endpoints.